decomposition.tex

```diff
@@ -137,21 +137,21 @@
     // addressability
     { &t } -> Same<T*>;
```

Comment:
The type of the address of a `const T` is a `const T*`. This raises an issue with your blanket wording about requirements and the const/r-/l-value convention. Sometimes the cv qualification or value category changes the return type in a way that effects the correctness of the constraint as written. I can't think of a clever way to handle this in general. The cases need to be spammed out.